### PR TITLE
Add copr guide for packaging upstream centos6 rpms.

### PR DESCRIPTION
**Pull request by jrafanie**
Moved a private wiki page to the guides repo.

Note, we'll need to convert this to use centos7 as have examples we need to package for 7.

@abellotti @carbonin Please review

### CODE REVIEW

File: developer_copr_setup.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+* Also, group rpms based upon how the yum repository will be consumed.
+* Name the project correctly since you can't rename it.
+
+### Configure the project's settings:
+
+* [chroot](https://help.ubuntu.com/community/BasicChroot):
+  * **epel-x86_64**
+* Edit this chroot and "specify additional packages to be always present in minimal buildroot":
+  * If you need to build a ruby gem that utilizes the SCL, you need the following:
+    * **scl-utils-build ruby193-build**
```
Comment (carbonin) [inline developer_copr_setup.md]: Should this be ruby 2.2.3?

File: developer_copr_setup.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    * **Note**: the guides suggest scl-utils, not scl-utils-build.
+      * Without scl-utils-build, some scl macros won't be found and the mock/build may fail.
+      * Bundler, rake-compiler, and others failed when using scl-utils: ```Unknown tag: %scl_package rubygem-%{gem_name}```
+
+* repositories:
+  * http://mirror.centos.org/centos/6/os/$basearch/
+  * http://mirror.centos.org/centos/6/SCL/$basearch/
+  * http://mirror.centos.org/centos/6/updates/$basearch/
```
Comment (carbonin) [inline developer_copr_setup.md]: Change to centos 7?

File: developer_copr_setup.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  * http://copr-be.cloud.fedoraproject.org/results/jrafanie/manageiq-scl/epel-6-$basearch/
+    * **Note**: The last repository is the project's yum repository itself since some packages have dependencies on other packages in the yum repository.
+    * For example, the qpid_messaging gem requires the qpid-cpp package be built first, so we build qpid-cpp and then qpid_messaging can be built using the resulting qpid-cpp rpms in this yum repository.
+
+### Find the desired srpm or create it.
+
+* If you have the desired package in distgit, you can use rhpkg/fedpkg srpm to produce an srpm you can then push to github or somewhere publicly.
+* Copr can use any src.rpm that is on the web.
+* Github is not required but that's what CFME people are familiar with.
```
Comment (carbonin) [inline developer_copr_setup.md]: CFME => MIQ

File: developer_copr_setup.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  * **Note** When you build in copr, you will need to provide the URL that it can be retrieved via curl/wget:
+    * To use: "rubygem-somegem-1.0.0-1.el6.src.rpm" found in: https://github.com/USERNAME/srpms/tree/master/srpms
+    * Tell mock/copr to use: https://raw.githubusercontent.com/USERNAME/srpms/master/srpms/rubygem-somegem-1.0.0-1.el6.src.rpm
```
Comment (carbonin) [inline developer_copr_setup.md]: el7 rather than el6

File: developer_copr_setup.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+### Configure mock to test builds locally
+
+Copr is sometimes busy and response can be slow. Using mock, to do test builds locally, can help avoid wasting time waiting for Copr to report the build failed. So use mock locally to debug builds until you get it to build successfully.
+
+* Create a /etc/mock/YOUR_PROJECT.cfg, by copying your target platform's .cfg.
+  * For example, I need epel-6-x86_64, so I do this:
```
Comment (carbonin) [inline developer_copr_setup.md]: Same here epel-7 rather and epel-6

**Comment by carbonin** [comment]
Should we update the examples and repos to reference Centos 7 rather then Centos 6?

**Comment by jrafanie** [comment]
> Should we update the examples and repos to reference Centos 7 rather then Centos 6?

@carbonin Yes, we should update it to centos 7.  I'm not sure if @abellotti ran through this with doing the wmi package but nevertheless, it's really just a guide we can update when have hands-on experience making it work with centos 7.

File: developer_copr_setup.md (hunk 3 above)
Comment (jrafanie) [inline developer_copr_setup.md]: Thanks, I'll fix this.

File: developer_copr_setup.md (hunk 1 above)
Comment (jrafanie) [inline developer_copr_setup.md]: Maybe, we'll need to fix this if we build anything that requires ruby.  I"m not even sure the exact SCL prefix name.  I guess it's ruby22 but I don't know.

**Comment by jrafanie** [comment]
@abellotti @carbonin Updated description so it's clear this is for centos6 and we'll need to run through this and update it as we do any packaging for centos7/ruby2.2.

File: README.md [hunk 6]
```diff
@@ -4,6 +4,7 @@
 * [Coding Style and Standards](coding_style_and_standards.md)
 * [Developer Setup](developer_setup.md)
 * [Development Appliance Setup](https://github.com/ManageIQ/manageiq-appliance-dev-setup)
+* [Developer copr setup](https://github.com/ManageIQ/developer_copr_setup.md)
```
Comment (Fryguy) [inline README.md]: COPR should be all caps, right?

File: developer_copr_setup.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
+## Developer Copr Setup:
```
Comment (Fryguy) [inline developer_copr_setup.md]: Same.  Also, no need for a `:` in a title

File: developer_copr_setup.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+* specify the chroot
+* add any additional packages to the minimal chroot
+* specify the `src.rpm` to build
+* host existing source rpm (srpm) publicly
+* mock, test, and build the rpm
+* incorporate the copr yum repo into our upstream community build.
+
+
+###  Create a fedora account (used as a copr account)
+* Create a fedora account: http://copr.fedoraproject.org/
```
Comment (Fryguy) [inline developer_copr_setup.md]: The grammar bug in me is dying from the lack of capitalization of proper nouns :D

File: README.md (hunk 6 above)
Comment (jrafanie) [inline README.md]: I don't think so, https://copr.fedoraproject.org/coprs/, it's only the first character that's capitalized.

File: developer_copr_setup.md (hunk 8 above)
Comment (jrafanie) [inline developer_copr_setup.md]: Fedora is a hat.

Comment (jrafanie) [inline developer_copr_setup.md]: :trollface:

File: developer_copr_setup.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+* Create a fedora / copr account
+* Create a copr project
+* add repositories for dependencies
+* specify the chroot
+* add any additional packages to the minimal chroot
+* specify the `src.rpm` to build
+* host existing source rpm (srpm) publicly
+* mock, test, and build the rpm
+* incorporate the copr yum repo into our upstream community build.
```
Comment (Fryguy) [inline developer_copr_setup.md]: Can these all be links down the respective sections?

File: README.md (hunk 6 above)
Comment (Fryguy) [inline README.md]: :+1:  Then it should be that way in our docs too.

Comment (jrafanie) [inline README.md]: Ah, yes...  good point...

File: developer_copr_setup.md (hunk 7 above)
Comment (abellotti) [inline developer_copr_setup.md]: Unless we're changing all links/basearchs/mock examples to 7, let's update the header to "Developer Copr Setup for CentOS 6x appliances" to minimize confusion for now.

**Comment by jrafanie** [comment]
Ok, I think I addressed @Fryguy @abellotti's comments.  We can do centos7 when we actually do it.

**Pull request merged**